fe/fe-core/src/main/java/com/starrocks/sql/optimizer/cost/CostModel.java

```diff
@@ -113,6 +119,40 @@ public static double getRealCost(CostEstimate costEstimate) {
                 costEstimate.getNetworkCost() * networkCostWeight;
     }
 
+    private static double adjustCostForMV(double realCost, ExpressionContext expressionContext) {
+        if (expressionContext.getOp() instanceof PhysicalOlapScanOperator) {
+            PhysicalOlapScanOperator scanOperator = expressionContext.getOp().cast();
+            // only adjust cost for mv scan operator
+            if (scanOperator.getTable() != null && scanOperator.getTable().isMaterializedView()) {
+                Statistics groupStatistics = expressionContext.getGroupStatistics();
+                Statistics mvStatistics = expressionContext.getStatistics();
+                // only adjust cost for mv scan operator when group statistics is unknown and mv group expression
+                // statistics is not unknown
+                if (groupStatistics != null && groupStatistics.getColumnStatistics().values().stream().
+                        anyMatch(ColumnStatistic::isUnknown) && mvStatistics.getColumnStatistics().values().stream().
+                        noneMatch(ColumnStatistic::isUnknown)) {
+                    Group group = expressionContext.getGroupExpression().getGroup();
+                    List<Double> costs = Lists.newArrayList();
+                    // get the costs of all expression in this group
+                    for (Pair<Double, GroupExpression> pair : group.getAllBestExpressionWithCost()) {
+                        if (!(pair.second.getOp() instanceof PhysicalOlapScanOperator)) {
+                            costs.add(pair.first);
+                        }
+                    }
+                    if (costs.size() > 0) {
+                        double groupMinCost = Collections.min(costs);
+                        if (realCost < groupMinCost) {
+                            return realCost;
+                        }
+                        double adjustCost = mvStatistics.getOutputRowCount();
+                        return Math.min(realCost, adjustCost);
```

Comment:
can this ensure less than `groupMinCost `?

Reply:
Fixed